src/stream/src/executor/backfill/cdc/cdc_backfill.rs

```diff
@@ -702,6 +787,103 @@ impl<S: StateStore> CdcBackfillExecutor<S> {
     }
 }
 
+struct UpstreamMessageBuffer {
```

Comment:
The implementation seems to be inspired by snapshot backfill, which is an overkill and more complicated than I thought.  

Is there any ordering requirement between table reader creation and polling upstream? In other words, can we make sure the offset we consume from upstream is accurate if we blindly poll upstream when the table reader is still under creation. If we can blindly poll upstream, how about just spawning a task to create the table reader with retry? Pseudo code example:
```rust
async fn create_table_reader_with_retry(...) -> TableReader  {
  ...
}

-- execute_inner
let handle = tokio::spawn(create_table_reader_with_retry(...));
loop {
  select! {
      biased;
      reader = handle => {
        ...
        break;
      }
      msg = upstream.next() => {
          // Emit barrier and skip chunk
     }
  }
```